### PR TITLE
Add us_ticker initialisation in MbedTester class

### PR DESCRIPTION
**Pull request by ithinuel**
### Description 

`wait_us` from `bootstrap/mbed_wait_api_no_rtos.c` will rely on the `us_ticker` if it is available on the platform.
In such case, `MbedTester` must initialize that ticker before using `wait_us`.

Proof is visible in https://github.com/ARMmbed/MCU-Driver-Arm/pull/7 not crashing.

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [ ] Feature update (New feature / Functionality change / New API)
    [ ] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test coverage 
 

    [x] Manual testing (and evidence provided)
    [ ] No additional tests required for this change (e.g documentation update)

### CODE REVIEW

File: tools/fpga_ci_test_shield/source/MbedTester.cpp [hunk 1]
```diff
@@ -586,6 +590,9 @@ MbedTester::MbedTester(const PinList *form_factor, const PinList *exclude_pins)
       _clk_index(MbedTester::physical_nc), _mosi_index(MbedTester::physical_nc), _miso_index(MbedTester::physical_nc), _aux_index(MbedTester::physical_nc),
       _clk(NULL), _mosi(NULL), _miso(NULL), _aux(NULL)
 {
+#ifdef DEVICE_USTICKER
```
Comment (fkjagodzinski) [inline tools/fpga_ci_test_shield/source/MbedTester.cpp:593]:
```suggestion
#ifdef DEVICE_USTICKER && !MBED_CONF_TARGET_INIT_US_TICKER_AT_BOOT
```

File: tools/fpga_ci_test_shield/source/MbedTester.cpp [hunk 2]
```diff
@@ -23,6 +23,10 @@
 #include "fpga_ci_test_shield/MbedTester.h"
 #include "utest/utest_print.h"
 
+#ifdef DEVICE_USTICKER
```
Comment (fkjagodzinski) [inline tools/fpga_ci_test_shield/source/MbedTester.cpp:26]:
```suggestion
#ifdef DEVICE_USTICKER && !MBED_CONF_TARGET_INIT_US_TICKER_AT_BOOT
```

**Review by fkjagodzinski — COMMENTED** [review]
Yep, the `bootstrap/mbed_wait_api.h` uses `hal/us_ticker_api.h` if `DEVICE_USTICKER` is set. This is missing the us ticker init at this point.

In mbed-os we used to rely on [`mbed_init()`](https://github.com/ARMmbed/mbed-os/blob/14e5d307bb6cdccb554b591ab2602d8d47e0b2d0/platform/source/mbed_sdk_boot.c#L92-L94) if `DEVICE_USTICKER` and `MBED_CONF_TARGET_INIT_US_TICKER_AT_BOOT` were set. The case for `MBED_CONF_TARGET_INIT_US_TICKER_AT_BOOT=0` was handled by the ticker middleware that has been removed from this repo.

I'm OK with letting this patch in, but this issue still requires a more generic solution. This patch only fixes the MbedTester class.

**Comment by fkjagodzinski** [comment]
A more complete solution requires updating [`bootstrap/mbed_boot.c`](https://github.com/MCU-Driver-HAL/MCU-Driver-HAL/blob/9f66a04985e5382e8306b0977c7a5f7a649eb7b4/bootstrap/mbed_boot.c#L125-L140) and every user of the us ticker (e.g. `bootstrap/mbed_wait_api_no_rtos.c`).

**Comment by fkjagodzinski** [comment]
Or even better, we can fix the `wait_us()` implementation to be completely independent from the us ticker by removing [
`#if DEVICE_USTICKER` case](https://github.com/MCU-Driver-HAL/MCU-Driver-HAL/blob/9f66a04985e5382e8306b0977c7a5f7a649eb7b4/bootstrap/mbed_wait_api_no_rtos.c#L24-L60). This will let us also remove `MBED_CONF_TARGET_INIT_US_TICKER_AT_BOOT` from the codebase.

**Comment by ithinuel** [comment]
> Or even better, we can fix the wait_us() implementation to be completely independent from the us ticker by removing
#if DEVICE_USTICKER case. This will let us also remove MBED_CONF_TARGET_INIT_US_TICKER_AT_BOOT from the codebase.

This is likely to quietly break other things that require some CPU time in while waiting.
If an interrupt fires during the wait_us (eg uart interrupt) the timer will continue to run while the busy loop will be delayed.

File: tools/fpga_ci_test_shield/source/MbedTester.cpp [hunk 3]
```diff
@@ -23,6 +23,10 @@
...
+#include "us_ticker_api.h"
```
Comment (hugueskamba) [inline tools/fpga_ci_test_shield/source/MbedTester.cpp:27]:
```suggestion
#include "hal/us_ticker_api.h"
```

**Review by hugueskamba — APPROVED** [review]

**Review by MarceloSalazar — APPROVED** [review]

**Pull request merged**